src/EnergyPlus/PCMThermalStorage.cc

```diff
@@ -0,0 +1,635 @@
+                 ++compNum) {
+                state.dataPlnt->PlantLoop(this->usePlantLoc.loopNum)
+                    .LoopSide(this->usePlantLoc.loopSideNum)
+                    .Branch(this->usePlantLoc.branchNum)
+                    .Comp(compNum)
+                    .FlowPriority = DataPlant::LoopFlowStatus::NeedyAndTurnsLoopOn;
+            }
+            for (int compNum = 1; compNum <= state.dataPlnt->PlantLoop(this->sourcePlantLoc.loopNum)
+                                                 .LoopSide(this->sourcePlantLoc.loopSideNum)
+                                                 .Branch(this->sourcePlantLoc.branchNum)
+                                                 .TotalComponents;
+                 ++compNum) {
+                state.dataPlnt->PlantLoop(this->sourcePlantLoc.loopNum)
+                    .LoopSide(this->sourcePlantLoc.loopSideNum)
+                    .Branch(this->sourcePlantLoc.branchNum)
+                    .Comp(compNum)
+                    .FlowPriority = DataPlant::LoopFlowStatus::NeedyAndTurnsLoopOn;
+            }
+
+            // Reset state variables for the start of the new environment.
+            this->EnergyStored = this->TankCapacity * this->LatentHeat;
+            this->PercentCapacity = 100.0;
+
+            // Flag that the initialization for this environment is complete.
+            this->MyEnvrnFlag = false;
+        }
+
+        // When BeginEnvrnFlag is reset at the end of a sizing period, allow the initialization to occur again for the next environment.
+        if (!state.dataGlobal->BeginEnvrnFlag) {
+            this->MyEnvrnFlag = true;
+        }
+
+        // Real64 avail = this->AvailabilitySchedule->getCurrentVal();
+        //  Establish component flow requests on each side of the storage tank.  Use the
+        //  design mass flow rates computed above and apply availability scheduling.  Note
+        //  that SetComponentFlowRate expects a mass flow (kg/s), not a volumetric flow.  The
+        //  previous implementation passed the design volumetric flow rate directly which
+        //  resulted in very small mass flows and in some cases zero flow on the use side.  Here
+        //  we convert to mass flow and schedule it off when the availability schedule is zero.
+    }
+
+    void PCMStorageData::Calculate(EnergyPlusData &state, PlantLocation const &plantLoc)
+    {
+        // plantLoc is passed in but not referenced in the body → silence here
+        (void)plantLoc;
+        auto &useInlet = state.dataLoopNodes->Node(UseSideInletNode);
+        auto &useOutlet = state.dataLoopNodes->Node(UseSideOutletNode);
+        auto &plantInlet = state.dataLoopNodes->Node(PlantSideInletNode);
+        auto &plantOutlet = state.dataLoopNodes->Node(PlantSideOutletNode);
+
+        // Real64 avail = this->AvailabilitySchedule->getCurrentVal();
+        Real64 dt_seconds = state.dataHVACGlobal->TimeStepSys * 3600.0;
+
+        Real64 CpWater = 4180.0; // J/kg-C
```

Comment:
Another water property that should be pulled from the actual plant loop fluid props.